### PR TITLE
docs: Document `on_load()` detail

### LINKED ISSUE #1695: `on_load()` contents not analyzed by `R CMD check`
Learned painfully today. Is it worth mentioning this in `?on_load` ?

Reference: https://github.com/duckdblabs/duckplyr/pull/122.

### PR DESCRIPTION
**Pull request by krlmlr**
Closes #1695.

### CODE REVIEW

File: R/aaa.R [hunk 1]
```diff
@@ -69,6 +69,10 @@ the <- new.env(parent = emptyenv())
 #' `on_load()` is implemented via `.onLoad()` and requires
 #' `run_on_load()` to be called from that hook.
 #'
+#' The expressions inside `on_load()` do not undergo static analysis
+#' by `R CMD check`. Therefore, it is advisable to only use
+#' simple function calls inside `on_load()`.
```
Comment (lionel-) [inline R/aaa.R:74]: I think we could add support for on_load to other tools so I'd remove the latter part:

```suggestion
#' Note that the expressions inside `on_load()` do not undergo static analysis
#' by `R CMD check`.
```

Comment (krlmlr) [reply R/aaa.R:74]: I don't understand "support ... to other tools", I'm fine with the change if it makes sense to you.

Comment (lionel-) [reply R/aaa.R:74]: On the other hand extracting to a function also makes sense in terms of style. Let's just merge as is.

**Comment by lionel-** [comment]
Thanks!

**Pull request merged**